ocs_ci/ocs/bucket_utils.py

```diff
@@ -139,6 +140,23 @@ def retrieve_test_objects_to_pod(podobj, target_dir):
     return downloaded_objects
 
 
+@retry(UnexpectedBehaviour, tries=5, delay=10, backoff=1)
+def check_objects_in_bucket(bucket_name, objects_list, mcg_obj, s3pod):
+    """
+    Checks object list present in bucket and compare it with uploaded object Lists
+    """
+    obj_list = list_objects_from_bucket(
+        s3pod,
+        f"s3://{bucket_name}",
+        s3_obj=mcg_obj,
+    )
+    if objects_list == obj_list:
```

Comment:
The function is expected to return True if the bucket contains all the objects in a given list, but this only checks for exact matches this function will return False if the bucket has an additional object that is not in the given list.

Using python sets gives us an easy solution:

```suggestion
    if set(objects_list).issubset(obj_list):
```

Reply:
Fixed